source/features/useful-not-found-page.tsx

```diff
@@ -36,54 +38,59 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
-	const parts = parseCurrentURL();
-	parts[2] = 'commits';
-	const url = '/' + parts.join('/');
-	if (await is404(location.origin + url)) {
+async function getLatestChangeToFile(): Promise<Record<string, any> | void> {
+	const url = new GitHubURL(location.href);
+	const {branch, filePath} = url;
+	if (!branch || !filePath) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			See also the {getType()}’s <a href={url}>commit history</a>.
-		</p>,
-	);
-}
-
-// If the resource exists in the default branch, link to it
-async function addDefaultBranchLink(bar: Element): Promise<void> {
-	const parts = getCleanPathname().split('/');
-	const branch = parts[3];
-	if (!branch) {
+	const commitsResponseObject = await api.v4(`
+		repository() {
+			ref(qualifiedName: "${branch}") {
+				target {
+					... on Commit {
+						history(first: 1, path: "${filePath}") {
```

Comment:
Every time I see this I wonder if it can can done in fewer steps. Can you look for a better solution? It probably doesn't exist, but…

Reply:
Sure, I'll try my best

Reply:
I've been messing around with the API and I wasn not able retrieve a list of changed files unless I specifically retrieved a commit using its `sha`/`oid`. I could ask on SO, if someone has a better solution.

Reply:
It may be possibly by using `object()` instead of `ref().target`

https://stackoverflow.com/a/55509589/288906

Reply:
It worked, thanks!